paymail_service_provider.go

```diff
@@ -157,6 +157,10 @@ func (p *PaymailDefaultServiceProvider) RecordTransaction(ctx context.Context,
 	metadata[p2pMetadataField] = p2pTx.MetaData
 	metadata[ReferenceIDField] = p2pTx.Reference
 
+	var rts recordIncomingTxStrategy
```

Comment:
You validate empty strategy here. Do not create a new structure instance, just call `rts.Validate()` after `getIncomingTxRecordStrategy(ctx, p.client, p2pTx.Hex)` (of course after error check)

Reply:
Changed the order